addons/mail/models/discuss/ir_config_parameter.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import re
+
+from odoo import api, models
+from odoo.exceptions import ValidationError
+
+
+class IrConfigParameter(models.Model):
+    _inherit = 'ir.config_parameter'
+
+    @api.constrains("value")
+    def _validate_link_preview_regexp(self):
+        for record in self:
+            if record.key == "mail.mail_link_preview_ignore_list":
+                for pattern in record.value.split("\n"):
+                    try:
+                        re.compile(pattern)
+                    except re.error:
+                        raise ValidationError(f"Invalid element: {pattern}")
```

Comment:
This should be translatable. Also the message should be a little bit less generic (why is it invalid?).
It feels like the original re.error might be more clear/useful to resolve the issue.

Reply:
You may use the "raise ... from" construct also, to have more insights.